c/libzsync/zsync.c

```diff
@@ -571,24 +567,18 @@ off_t zsync_file_length(struct zsync_state *zs) {
  * Tell libzsync to move the local copy of the target (or under construction
  * target) to the given filename. */
 int zsync_rename_file(struct zsync_state *zs, const char *f) {
-    char *rf = zsync_cur_filename(zs, f);
-
-    if (strcmp(zs->cur_filename, f)) {
-        close(rcksum_filehandle(zs->rs));
-        int x = unlink(rf);
-        if (!x) {
-            free(rf);
-            zs->cur_filename = strdup(f);
-            rcksum_fileopen(zs->rs);
-        }
-        else
-            perror("unlink");
+    char *rf = zsync_cur_filename(zs);
```

Comment:
minor, not worth changing: I think rcksum shouldn't know the filename in the first place, it should just get a fd. Then this odd dance of moving the filename out of rcksum and into zsync wouldn't be necessary.